src/modules/V3ToV4Migrator.sol

```diff
@@ -21,4 +32,68 @@ abstract contract V3ToV4Migrator is RouterImmutables {
         return caller == owner || V3_POSITION_MANAGER.getApproved(tokenId) == caller
             || V3_POSITION_MANAGER.isApprovedForAll(owner, caller);
     }
+
+    /// @dev check that the v4 position manager call is a safe call
+    /// of the position-altering Actions, we only allow Actions.MINT
+    /// this is because, if a user could be tricked into approving the UniversalRouter for
+    /// their position, an attacker could take their fees, or drain their entire position
+    function _checkV4ClPositionManagerCall(bytes calldata inputs) internal view {
+        bytes4 selector;
+        assembly {
+            selector := calldataload(inputs.offset)
+        }
+        if (selector != V4_CL_POSITION_MANAGER.modifyLiquidities.selector) {
+            revert InvalidAction(selector);
+        }
+
+        // slice is `abi.encode(bytes unlockData, uint256 deadline)`
+        bytes calldata slice = inputs[4:];
+        // the first bytes(0) extracts the unlockData parameter from modifyLiquidities
+        // unlockData = `abi.encode(bytes actions, bytes[] params)`
+        // the second bytes(0) extracts the actions parameter from unlockData
+        bytes calldata actions = slice.toBytes(0).toBytes(0);
+
+        uint256 numActions = actions.length;
+
+        for (uint256 actionIndex = 0; actionIndex < numActions; actionIndex++) {
+            uint256 action = uint8(actions[actionIndex]);
```

Comment:
Why casting uint8 here for a uint256 variable?

Reply:
`actions[actionIndex]` returns a bytes1 and we need to explictly convert it to a `uint` type